### PR TITLE
genesets route for local server

### LINKED ISSUE #2044: Rewrite geneset route backend after hosted / desktop divorce is complete
From Bruce: Re-create once Marcus lands his divorce PR.  I recommend we create a separate PR for the back-end, so Colin's branch/PR is just front-end.

### LINKED ISSUE #2045: Implement gene sets desktop CLI params and CLI error handling
Requirements documented here: https://docs.google.com/document/d/1O-Os0Bg6a9FBKAA1KIMR12KESAEhj2pD32lJFC5OGko/edit#heading=h.a1yl02wtw6p9

### LINKED ISSUE #2048: Implement read/write layer for file format
See file format requirements in 1F [here](https://docs.google.com/document/d/1O-Os0Bg6a9FBKAA1KIMR12KESAEhj2pD32lJFC5OGko/edit#)

### LINKED ISSUE #2063: Implement genesets REST API
Requirements here: https://docs.google.com/document/d/12Wce-r2OvLGnuUWm9abAiZuj-kELz_LYTEvv6KZLXXo/edit#heading=h.wm3l0scc2cy5

### PR DESCRIPTION
**Pull request by bkmartinjr**
This PR contains an subset of back-end geneset support for the _local_ server only:
1. PUT and GET operations for the /genesets route
2. Geneset CSV persistence
3. Unit tests
4. Config route changes to support genesets

This PR does not contain geneset summary route support or front-end support, which will arrive in additional PRs.

Fixes #2063 
Fixes #2048 
Fixes #2044 
Fixes #2045

### CODE REVIEW

**Review by colinmegill — APPROVED** [review]

File: local_server/test/fixtures/pbmc3k-genesets.csv [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+first geneset name,,F5, a gene_description
+first geneset name,a description, NO_SUCH_GENE, non-existent gene
+first geneset name,a description, F5, duplicate gene
+first geneset name, a description, SUMO3,
+first geneset name,, SRM,
+second geneset,,RER1
+second geneset,,SIK1
+third geneset,,NO_SUCH_GENE
+fourth_geneset,fourth description,,gene intentionally missing
+fifth_dataset,,,
```
Comment (MDunitz) [inline local_server/test/fixtures/pbmc3k-genesets.csv:12]: is this intentional? or should it be fifth_geneset?

Comment (MDunitz) [inline local_server/test/fixtures/pbmc3k-genesets.csv:12]: should this be fifth_geneset?

**Review by MDunitz — APPROVED** [review]
lgtm!

File: local_server/test/fixtures/pbmc3k-genesets.csv (hunk 1 above)
Comment (bkmartinjr) [reply local_server/test/fixtures/pbmc3k-genesets.csv:12]: The exact string doesn't really matter as it is just a name - as long as the test "expects" the same, all will be fine.  But yeah, I had a brain fault when I typed it :-)  lolz.

**Pull request merged**